components/ProjectForm/ProjectForm.js

```diff
@@ -55,40 +57,83 @@ const Select = styled.select`
   box-shadow: 1px 1px 1px 1px rgb(204 203 203);
   margin-left: 30px;
 `;
-export default function ProjectForm({ onAddProject, onCloseForm }) {
-  const [imageFile, setImageFile] = useState(null);
+
+export default function ProjectForm({ onAddProject }) {
   const [isSelectOpen, setIsSelectOpen] = useState(false);
-  const handleSubmit = (event) => {
-    event.preventDefault();
-    const formData = new FormData(event.target);
-    const data = Object.fromEntries(formData);
-    if (imageFile) {
-      const reader = new FileReader();
-      reader.onload = () => {
-        data.imageSource = reader.result;
-        onAddProject(data);
-      };
-      reader.readAsDataURL(imageFile);
-    } else {
-      data.imageSource = "/placeholder-image.jpg";
-      onAddProject(data);
-    }
-    event.target.reset();
-    alert("You have successfully submitted your project!");
-    onCloseForm();
-  };
+  const [projects, setProjects] = useLocalStorageState("projects", {
+    defaultValue: [],
+  });
+  const [imageFile, setImageFile] = useState(null);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [error, setError] = useState(null);
+
   const handleImageChange = (event) => {
-    const file = event.target.files[0];
-    setImageFile(file);
+    setImageFile(event.target.files[0]);
```

Comment:
You could add some error handling here, if e.g. the use selected an invalid file.